packages/rearch/lib/rearch.dart

```diff
@@ -225,6 +228,28 @@ class CapsuleContainer implements Disposable {
       manager.dispose();
     }
   }
+
+  /// Warms up the specified capsules by waiting for them to all be [AsyncData].
+  Future<void> warmUp<T extends AsyncValue<dynamic>>(
+    Iterable<Capsule<T>> capsules,
+  ) async {
+    final completer = Completer<()>();
+    final dispose = listen((use) {
+      final resolved = capsules.map((capsule) => use(capsule));
+
+      if (resolved.every((capsule) => capsule is AsyncData)) {
+        completer.complete(());
+      }
+
+      final asyncError = resolved.whereType<AsyncError<dynamic>>().firstOrNull;
```

Comment:
you should just be able to do `whereType<AsyncError>`, or if that throws a fit with the dart analyzer, `whereType<AsyncError<T>>`